File: rules-interpreter/src/main/java/io/github/iamrenny/ruleflow/evaluators/BinaryAndContextEvaluator.java
package io.github.iamrenny.ruleflow.evaluators;

import io.github.iamrenny.ruleflow.RuleFlowLanguageParser;
import io.github.iamrenny.ruleflow.visitors.Visitor;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class BinaryAndContextEvaluator implements ContextEvaluator<RuleFlowLanguageParser.BinaryAndContext> {
    private static final Logger logger = LoggerFactory.getLogger(BinaryAndContextEvaluator.class);

    @Override
    public Boolean evaluate(RuleFlowLanguageParser.BinaryAndContext ctx, Visitor visitor) {
        boolean b = (Boolean) visitor.visit(ctx.left) && (Boolean) visitor.visit(ctx.right);
        logger.debug("BinaryAnd: left={}, right={}, result={}", ctx.left, ctx.right, b);
        return b;
    }
}

File: rules-interpreter/src/main/kotlin/com/rappi/fraud/rules/parser/evaluators/BinaryAndContextEvaluator.kt
package com.rappi.fraud.rules.parser.evaluators

import com.rappi.fraud.analang.ANALexer
import com.rappi.fraud.analang.ANAParser
import com.rappi.fraud.rules.parser.visitors.Visitor

class BinaryAndContextEvaluator : ContextEvaluator<ANAParser.BinaryAndContext> {

    override fun evaluate(ctx: ANAParser.BinaryAndContext, visitor: Visitor): Boolean {
        return visitor.visit(ctx.left) as Boolean && visitor.visit(ctx.right) as Boolean
        }
}